src/main/java/dk/creditoro/client/view/frontpage/FrontpageController.java

```diff
@@ -98,9 +103,16 @@ public void btnProductions(ActionEvent actionEvent) {
      */
     public void onSearchAction(ActionEvent actionEvent) {
         LOGGER.info("Hvad f skal der ske med den her søgefunktion");
-        frontpageViewModel.search();
+        var view = choiceBox.getSelectionModel().getSelectedItem();
+        if (view.equals("Kanal")) {
+            viewHandler.openView(Views.BROWSE_CHANNELS, searchTextField.getText());
+
+        } else {
+            viewHandler.openView(Views.BROWSE_PRODUCTIONS, searchTextField.getText());
+        }
```

Comment:
Call .clear() on searchTextField when enter is pressed. Otherwise the search query will still be present when you come back to the frontpage.
```suggestion
        }
        searchTextField.clear();
```